README.md

```diff
@@ -41,42 +41,42 @@ message HelloWorldResponse {
 ```
 
 Run the `protoc` binary to generate `gen/haberdasher_pb.rb` and `gen/haberdasher_twirp.rb`.
-```
+```sh
 ➜ protoc --proto_path=. ./haberdasher.proto --ruby_out=gen --twirp_ruby_out=gen
 ```
 
 Write an implementation of our haberdasher service and attach to a rack server:
-```
+```ruby
 # main.rb
 require 'rack'
 require_relative 'gen/haberdasher_pb.rb'
 require_relative 'gen/haberdasher_twirp.rb'
 
-class HaberdasherImplementation
-    def HelloWorld(req)
-        return Examples::HelloWorldResponse.new(message: "Hello #{req.name}")
+class HaberdasherHandler
+    def hello_world(req)
+        return Example::HelloWorldResponse.new(message: "Hello #{req.name}")
     end
 end
 
-svc = HaberdasherImplementation.new()
-Rack::Handler::WEBrick.run HaberdasherService.new(svc).handler
+svc = HaberdasherHandler.new()
+Rack::Handler::WEBrick.run Proc.new {|env| Example::HaberdasherService.new(svc).call(env)}
 ```
 
 You can also mount onto a rails service:
-```
+```ruby
 App::Application.routes.draw do
   svc = HaberdasherImplementation.new()
-  mount HaberdasherService.new(svc).handler, at: HaberdasherService::PATH_PREFIX
+  mount Proc.new {|env| Example::HaberdasherService.new(svc).call(env)}, at: HaberdasherService::PATH_PREFIX
```

Comment:
ah, no need to make a new Proc with a block that calls call ... lol. Implementing `.call` is what makes the service a valid Rack middleware block. Also need to add the module prefix to the path prefix part.

Change the example to this:
```ruby
handler := HaberdasherImplementation.new
service := Example::HaberdasherService.new(handler)
App::Application.routes.draw do
  mount service, at: Example::HaberdasherService::PATH_PREFIX
end
```

Reply:
oh derp